### PR TITLE
publishing diagnostics for all files #823

### PR DESCRIPTION
**Pull request by nehashri**
All files are validated to publish diagnostics.

Diagnostics are published when
- when gauge lsp server is initialized
- a gauge file(.spec and .cpt) is opened
- any file is changed (gauge or implementation files)

A lock on publish diagnostics makes sure only one go routine performs this at a time.

Suggestions are added to duplicate step validation errors. They are filtered before printing suggestions on console.

### CODE REVIEW

File: validation/suggest.go [hunk 1]
```diff
@@ -31,13 +31,26 @@ func showSuggestion(validationErrors validationErrors) {
 	if !HideSuggestion {
 		for t, errs := range groupErrors(validationErrors) {
 			fmt.Println(getSuggestionMessage(t))
-			for _, err := range errs {
-				fmt.Println(err.Suggestion())
+			suggestions := filterDuplicateSuggestions(errs)
```
Comment (kashishm) [inline validation/suggest.go]: When would there be duplicate suggestions?

Comment (nehashri) [reply validation/suggest.go]: We have duplicate suggestions printed if there is an unimplemented step is used multiple times.

Comment (kashishm) [reply validation/suggest.go]: I just tried it, it is printing without the duplicate suggestions.

Comment (nehashri) [reply validation/suggest.go]: In master the suggestions was added to only one error. The duplicate errors did not have the suggestions. But this doesn't work well for lsp diagnostics. We need the suggestions on all errors. Hence, there are changes in validate.go where we copy over the suggestions while duplicating the errors. 
After this change, we had to filter the duplicate suggestion while printing on the console.

**Pull request merged**